robot_smach_states/src/robot_smach_states/util/designators/core.py

```diff
@@ -26,25 +26,41 @@ def __init__(self, initial_value=None, resolve_type=None, name=None):
         """
         Initialization method
 
-        :param initial_value: () initial value
-        :param resolve_type: (type) type to which this designator should resolve
-        :param name: (str) name used for debugging purposes
+        :param initial_value: Initial value, should match the given resolve_type.
+                              None is allowed if a resolve_type is provided.
+        :param resolve_type: Type to which this designator should resolve.
+                             If None, use the type of the initial value. Value
+                             must be not None in that case.
+        :vartype resolve_type: Type or a list with one type (the element type).
+
+        :param name: name used for debugging purposes
+        :vartype name: str
         """
         super(Designator, self).__init__()
 
         self._name = name
-
         self.__initial_value = initial_value
-        if not resolve_type:
+        if resolve_type:
+            if isinstance(resolve_type, list):
+                if len(resolve_type) != 1:
```

Comment:
I think there are cases where a list of types is used as 'either one of these types' but if so (and I'm not sure), then _that_ use-case is just wrong.

Reply:
Clarified semantics of resolve_type somewhat

Reply:
The current implementation allows the following: `[[float], [int], float, int]`. So multiple resolve types are allowed. If one of the resolve types is a list, that list should contain all of the same types, so the `len([float]) == 1`.

Reply:
In my changes, a single type means a value of that type, and list with a single type means a list of values of that type.
This is not really nice, but Python lacks the concept "list of <type>" in its type system.

As Loy says and I agree, there should not be a set of allowed types. Inheritance is supported though as it uses `isinstance`. That handles `int` vs `float` already. Mixing single values and lists of values is non-Pythonic, you should know what you get without code to figure that out.

Reply:
Another option, if we want to allow multiple types, is to pass that asa `set()`, which is also a builtin type. 
But I do prefer to just allow 1 type to be specified. 

So, this change if fine AFAIK.

Reply:
Right now a mix of lists and single items is allowed. Also different single items/items in the list are allowed. So Before changing this API, please check the entire `tue_robocup` for compatibility.

Reply:
I checked the designators in tue_robocup, and found no cases of having multiple types in the constructor call. If my check was incomplete, at least it is not a commonly used feature.